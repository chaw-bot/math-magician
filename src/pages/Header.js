import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Home',
      classname: 'homePage',
    },
    {
      id: 2,
      path: '/calculator',
      text: 'Calculator',
      classname: 'calcPage',
    },
    {
      id: 2,
      path: '/quotes',
      text: 'Quotes',
      classname: 'quotePage',
    },
  ];

  return (
    <div className="nav">
      <nav className="navBar">
        <h1 className="title">
          MathMagicians
        </h1>
        <ul className="otherPages">
          {links.map((link) => (
            <li className={link.classname} key={link.id}>
              <NavLink to={link.path} activeClassName="active-link" exact>{link.text}</NavLink>
            </li>
          ))}
        </ul>
      </nav>
    </div>
  );
};

export default Header;
