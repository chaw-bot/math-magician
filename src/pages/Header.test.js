import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from './Header';

it('Render Header', () => {
  const tree = renderer
    .create(<Router><Header /></Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
