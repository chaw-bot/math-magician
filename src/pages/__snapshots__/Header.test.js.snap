// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header 1`] = `
<div
  className="nav"
>
  <nav
    className="navBar"
  >
    <h1
      className="title"
    >
      MathMagicians
    </h1>
    <ul
      className="otherPages"
    >
      <li
        className="homePage"
      >
        <a
          aria-current="page"
          className="active-link"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
      </li>
      <li
        className="calcPage"
      >
        <a
          aria-current={null}
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="quotePage"
      >
        <a
          aria-current={null}
          href="/quotes"
          onClick={[Function]}
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
  <footer
    className="footer"
  >
    <p>
      Created by Chawanzi Ng'uni
    </p>
  </footer>
</div>
`;
